src/components/views/right_panel/UserInfo.js

```diff
@@ -0,0 +1,1212 @@
+            // not a DM room if they are not joined
+            if (!them || !them.membership || them.membership !== 'join') continue;
+
+            const highlight = room.getUnreadNotificationCount('highlight') > 0;
+
+            tiles.push(
+                <RoomTile key={room.roomId}
+                          room={room}
+                          transparent={true}
+                          collapsed={false}
+                          selected={false}
+                          unread={Unread.doesRoomHaveUnreadMessages(room)}
+                          highlight={highlight}
+                          isInvite={false}
+                          onClick={onRoomTileClick}
+                />,
+            );
+        }
+    }
+
+    const labelClasses = classNames({
+        mx_UserInfo_createRoom_label: true,
+        mx_RoomTile_name: true,
+    });
+
+    let body = tiles;
+    if (!body) {
+        body = (
+            <AccessibleButton className="mx_UserInfo_createRoom" onClick={onNewDMClick}>
+                <div className="mx_RoomTile_avatar">
+                    <img src={require("../../../../res/img/create-big.svg")} width="26" height="26" alt={_t("Start a chat")} />
+                </div>
+                <div className={labelClasses}><i>{ _t("Start a chat") }</i></div>
+            </AccessibleButton>
+        );
+    }
+
+    return (
+        <div className="mx_UserInfo_container">
+            <div className="mx_UserInfo_container_header">
+                <h3>{ _t("Direct messages") }</h3>
+                <AccessibleButton
+                    className="mx_UserInfo_container_header_right mx_UserInfo_newDmButton"
+                    onClick={onNewDMClick}
+                    title={_t("Start a chat")}
+                />
+            </div>
+            { body }
+        </div>
+    );
+});
+
+const UserOptionsSection = withLegacyMatrixClient(({matrixClient: cli, member, isIgnored, canInvite}) => {
+    let ignoreButton = null;
+    let insertPillButton = null;
+    let inviteUserButton = null;
+    let readReceiptButton = null;
+
+    const onShareUserClick = () => {
+        const ShareDialog = sdk.getComponent("dialogs.ShareDialog");
+        Modal.createTrackedDialog('share room member dialog', '', ShareDialog, {
+            target: member,
+        });
+    };
+
+    // Only allow the user to ignore the user if its not ourselves
+    // same goes for jumping to read receipt
+    if (member.userId !== cli.getUserId()) {
+        const onIgnoreToggle = () => {
+            const ignoredUsers = cli.getIgnoredUsers();
+            if (isIgnored) {
+                const index = ignoredUsers.indexOf(member.userId);
+                if (index !== -1) ignoredUsers.splice(index, 1);
+            } else {
+                ignoredUsers.push(member.userId);
+            }
+
+            cli.setIgnoredUsers(ignoredUsers).then(() => {
+                // return this.setState({isIgnoring: !this.state.isIgnoring});
```

Comment:
need to call `setIsIgnored` from here still I guess?

Reply:
To work around lack of local echo I guess so, the value actually comes down https://github.com/matrix-org/matrix-react-sdk/pull/3465/files#diff-a5813c8342fa7717bb0bcca63f59eacaR770

Reply:
aha, that's fine then. Can we remove the commented code?

Reply:
Can we still remove the commented code though?